subsys/shell/shell.c

```diff
@@ -704,6 +704,16 @@ static int execute(const struct shell *shell)
 
 	}
 
+	if ((cmd_lvl == CONFIG_SHELL_ARGC_MAX) && (argc == 2)) {
+		/* argc == 2 indicates that when command string was parsed
+		 * there was more characters remaining. It means that number of
+		 * arguments exceeds the limit.
+		 */
+		shell_internal_fprintf(shell, SHELL_ERROR,
```

Comment:
This is not an error but a shell feature described in the documentation. If there are more arguments than `CONFIG_SHELL_ARGC_MAX` remaining are treated as a 1 argument.

Reply:
discussed offline with @jakub-uC: it is a documented behavior of shell but it is probably not what user expects. Since if too many arguments is provided then it's better to get notified than pass unexpected arguments to the handler.

Reply:
removed information from kconfig.